mim/_internal/export/pack_cfg.py

```diff
@@ -0,0 +1,285 @@
+            'train.py', 'test.py', 'dist_train.sh', 'dist_test.sh',
+            'slurm_train.sh', 'slurm_test.sh'
+    ]:
+        pack_tools(
+            tool_name=tool_name,
+            scope=default_scope,
+            tool_dir=tools_dir,
+            auto_import=True)
+
+    # TODO: get demo.py
+
+    if not save_log:
+        shutil.rmtree(cfg['work_dir'])
+
+    dump()
+    return 0
+
+
+def keyboardinterupt_handler(sig: int, frame, export_root_dir: str):
+    """Clear uncompleted exported package by interrupting with keyboard."""
+    if osp.exists(export_root_dir):
+        shutil.rmtree(export_root_dir)
+
+    sys.exit(-1)
+
+
+def error_postprocess(export_root_dir: str, export_log_dir: str, cfg_name: str,
+                      error_key: str):
+    """Print Debug message when package can't successfully export for missing
+    datasets.
+
+    Args:
+        export_root_dir (str): _description_
+        absolute_cfg_path (str): _description_
+        origin_cfg (ConfigDict): _description_
+        error_key (str): _description_
+        logger (_type_): _description_
+    """
+    from mim.utils import echo_error
+    if osp.exists(export_root_dir):
+        shutil.rmtree(export_root_dir)
+
+    traceback.print_exc()
+
+    error_msg = f"{'=' * 20} Debug Message {'=' * 20}"\
+        f"\nThe data root of '{error_key}' is not found. You can"\
+        ' use the below two method to deal with.\n\n'\
+        "    >>> Method 1: Please modify the 'data_root' in"\
+        f" duplicate config '{export_log_dir}/{cfg_name}'.\n"\
+        "    >>> Method 2: Use '--model_only' to export model only.\n\n"\
+        "After finishing one of the above steps, you can use 'mim export"\
+        f" {export_log_dir}/{cfg_name} [--model-only]' to export again."
+
+    echo_error(error_msg)
+
+    sys.exit(-1)
+
+
+def pack_tools(tool_name: str,
+               scope: str,
+               tool_dir: str,
+               auto_import: Optional[bool] = False):
+    """pack tools from installed repo.
+
+    Args:
+        tool_name (str): Tool name in repos' tool dir.
+        scope (str): The scope of repo.
+        path (str): Path to save tool.
+        auto_import (bool, optional): Automatically add "import pack" to the
+            tool file. Defaults to "False"
+    """
+    pkg_root = get_installed_path(scope)
+    path = osp.join(tool_dir, tool_name)
+
+    if os.path.exists(path):
+        os.remove(path)
+
+    # tools will be put in package/.mim in PR #68
+    tool_script = osp.join(pkg_root, '.mim', 'tools', tool_name)
+    if not osp.exists(tool_script):
+        tool_script = osp.join(pkg_root, 'tools', tool_name)
+
+    shutil.copy(tool_script, path)
+
+    # automatically import the pack modules
+    if auto_import:
+        with open(path, 'r+') as f:
+            lines = f.readlines()
+            code = ''.join(lines[:1] + [_import_pack_str] + lines[1:])
+            f.seek(0)
+            f.write(code)
+            f.truncate()
```

Comment:
Why do we need to call `truncate` function here?

Reply:
It's no use now. I will delete it.